sources/Renderer/Vulkan/VKRenderSystem.cpp

```diff
@@ -961,6 +961,45 @@ VKDeviceBuffer VKRenderSystem::CreateStagingBufferAndInitialize(
     return stagingBuffer;
 }
 
+VKDeviceBuffer VKRenderSystem::CreateTextureStagingBufferAndInitialize(
+    const VkBufferCreateInfo&   createInfo,
+    const ImageView&            srcImageView,
+    const Extent3D&             extent)
+{
+    /* Allocate staging buffer */
+    VKDeviceBuffer stagingBuffer = CreateStagingBuffer(createInfo);
+
+    const void* data = srcImageView.data;
+    const std::size_t dataSize = srcImageView.dataSize;
+
+    /* Copy initial data to buffer memory */
+    if (data != nullptr && dataSize > 0) {
+        if (VKDeviceMemoryRegion* region = stagingBuffer.GetMemoryRegion())
+        {
+            /* Map buffer memory to host memory */
+            VKDeviceMemory* deviceMemory = region->GetParentChunk();
+            if (void* memory = deviceMemory->Map(device_, region->GetOffset(), dataSize))
+            {
+                if (srcImageView.stride > 0) {
+                    const std::uint32_t stride = GetMemoryFootprint(srcImageView.format, srcImageView.dataType, srcImageView.stride);
+                    const std::size_t width = GetMemoryFootprint(srcImageView.format, srcImageView.dataType, extent.width);
+
+                    for (std::uint32_t row = 0; row < extent.height; ++row) {
+                        ::memcpy(static_cast<std::uint8_t*>(memory) + row * width, static_cast<const std::uint8_t*>(data) + row * stride, width);
```

Comment:
You should be able to replace the `memcpy`s and this for-loop with `BitBlit()`, it's defined in _sources/Core/ImageUtils.h_ and can handle row and layer strides.